packages/core/src/lens.ts

```diff
@@ -0,0 +1,101 @@
+type Prev = [
```

Comment:
Could you add a brief introduction/a link on top of this file about lens mas @arrizalamin?

If I'm not mistaken it's something like this https://github.com/hatashiro/lens.ts?